pkg/dns/provider/zonecache.go

```diff
@@ -294,12 +294,17 @@ func (c *defaultZoneCache) deleteZoneState(zone DNSHostedZone) {
 	c.persistZone(zone)
 }
 
-func (c *defaultZoneCache) ApplyRequests(err error, zone DNSHostedZone, reqs []*ChangeRequest) {
+func (c *defaultZoneCache) ApplyRequests(logctx logger.LogContext, err error, zone DNSHostedZone, reqs []*ChangeRequest) {
 	if err == nil {
 		c.state.ExecuteRequests(zone, reqs)
 		c.persistZone(zone)
 	} else {
-		c.deleteZoneState(zone)
+		if !errors.IsThrottlingError(err) {
```

Comment:
Are you sure that in case of throttling errors the partly executed batch set(s) are correctly updated to the cache?

Reply:
The AWS provider only returns a ThrottlingError if all batches have been rejected with code `Throttling`.
In this case no update of the cache is needed, as nothing was executed.
See https://github.com/gardener/external-dns-management/pull/166/files#diff-66fd0bbf40c6d7d2b6ace4f102de350b0f578464b5682d1ca34b2ab1791b7c71R154